File: src/commands/LSTreeCommand.ts
import { lsTree } from "../core/tree/lsTree.js";
import logger from "../common/helpers/logger.js";
import { InvalidCommandError } from "../common/helpers/errors/minigit.js";
import { Command } from "./types.js";

export class LSTreeCommand implements Command {
  execute(args: string[]): void {
    const [treeHash] = args;

    if (!treeHash) {
      throw new InvalidCommandError("sha was not passed in command arguments");
    }

    const entries = lsTree(treeHash);
    logger.logTree(entries);
  }
}

File: src/cli/commands/ls_tree_command.cpp
#include <iostream>

#include "minigit/cli/commands/ls_tree_command.hpp"
#include "minigit/format/output.hpp"
#include "minigit/common/util/error.hpp"

namespace minigit::cli {

    int LsTreeCommand::execute(repo::Repository* repo, std::span<const std::string_view> args) const
    {
        if (args.empty())
        {
            throw Error(ErrorCode::InvalidCommand,
                        "INVALID_COMMAND_ERROR: sha was not passed in command arguments");
        }

        const auto entries = repo->ls_tree(model::ObjectId::from_hex(args[0]));
        format::print_tree(std::cout, entries);
        return 0;
    }

} // namespace minigit::cli
